pkg/store/consul/kv.go

```diff
@@ -226,6 +228,59 @@ func (c consulStore) DeletePod(podPrefix PodPrefix, nodename types.NodeName, pod
 	return writeMeta.RequestTime, nil
 }
 
+// TransformPod mutates the input context in such a way
+// that it will transactionally transform the pod manifest on the given nodes.
+// If the pod does not exist on some of the given nodes,
+// the operation skips those nodes without considering that an error.
+func (c consulStore) TransformPod(
```

Comment:
nit: we've used the word `Mutate` elsewhere e.g. https://github.com/square/p2/blob/master/pkg/store/consul/pcstore/consul_store.go#L230